src/Stratis.Bitcoin.IntegrationTests/Wallet/SendingToAndFromManyAddressesSpecification_Steps.cs

```diff
@@ -0,0 +1,189 @@
+                .StratisPowNode(NodeTwo).Start().NotInIBD().WithWallet(WalletName, WalletPassword)
+                .WithConnections()
+                    .Connect(NodeOne, NodeTwo)
+                    .AndNoMoreConnections()
+                .Build();
+        }
+
+        private void node1_sends_funds_to_node2_TO_fifty_addresses()
+        {
+            // Mine block
+            this.CoinBaseMaturity = (int)this.nodes[NodeOne].FullNode.Network.Consensus.Option<PowConsensusOptions>().CoinbaseMaturity;
+
+            this.sharedSteps.MineBlocks(this.CoinBaseMaturity + 50
+                , this.nodes[NodeOne]
+                , WalletAccountName
+                , WalletName
+                , WalletPassword);
+
+            // Sync both nodes.
+            this.nodes[NodeOne].CreateRPCClient().AddNode(this.nodes[NodeTwo].Endpoint, true);
+
+            this.sharedSteps.WaitForBlockStoreToSync(this.nodes[NodeTwo], this.nodes[NodeOne]);
+
+            // Get 50 unused addresses from node two.
+            IEnumerable<HdAddress> recevierAddresses = this.nodes[NodeTwo].FullNode.WalletManager()
+                .GetUnusedAddresses(new WalletAccountReference(WalletName, WalletAccountName), 50);
+
+            List<Recipient> recipients = recevierAddresses.Select(address => new Recipient
+            {
+                ScriptPubKey = address.ScriptPubKey,
+                Amount = Money.COIN
+            }).ToList();
+
+            this.transactionBuildContext = SharedSteps.CreateTransactionBuildContext(WalletName
+                , WalletAccountName
+                , WalletPassword
+                , recipients
+                , FeeType.Medium
+                , 101);
+
+            Transaction transaction = this.nodes[NodeOne].FullNode.WalletTransactionHandler().
+                BuildTransaction(this.transactionBuildContext);
+
+            transaction.Outputs.Count.Should().Be(51);
+
+            // Broadcast transaction to node two.
+            this.nodes[NodeOne].FullNode.NodeService<WalletController>().
+                SendTransaction(new SendTransactionRequest(transaction.ToHex()));
+
+            // Wait for the transactions to arrive.
+            TestHelper.WaitLoop(() => this.nodes[NodeTwo].CreateRPCClient().GetRawMempool().Length > 0);
+            TestHelper.WaitLoop(() => this.nodes[NodeTwo].FullNode.WalletManager().GetSpendableTransactionsInWallet(WalletName).Any());
+        }
+
+        private void node2_receives_the_funds()
+        {
+            this.stratisReceiverTotalAmount = this.nodes[NodeTwo].FullNode.WalletManager().
+                GetSpendableTransactionsInWallet(WalletName).
+                Sum(s => s.Transaction.Amount);
+
+            this.stratisReceiverTotalAmount.Should().Be(5000000000);
+
+            this.nodes[NodeTwo].FullNode.WalletManager().
+                GetSpendableTransactionsInWallet(WalletName).
+                First().Transaction.BlockHeight.
+                Should().BeNull();
```

Comment:
why should first transaction blockheight be null ?

Reply:
Because nodetwo is not in sync with nodeone.  Before and after test.